workflow-report/action.yml

```diff
@@ -0,0 +1,73 @@
+name: 'Send VK Teams notifications about failed job'
+description: 'This action create a message about failed job and send it to VK Teams chat'
+
+inputs:
+  api-url:
+    description: 'Bot API URL (ICQ/Myteam)'
+    required: false
+    default: 'https://api.internal.myteam.mail.ru/bot/v1/'
+  bot-token:
+    description: 'Bot token'
+    required: true
+  chat-id:
+    description: 'Notification chat ID (or stamp from chat URL)'
+    required: true
+  workflow-steps:
+    description: 'Get steps to show it in message'
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: Compile message about workflow failure
+      id: compile-message
+      uses: actions/github-script@v6
+      with:
+        script: |
+          const steps = ${{ inputs.workflow-steps }};
+          const event = `${{ github.event_name }}`;
+          const headRef = `${{ github.head_ref }}`;
+          const ref = `${{ github.ref }}`;
+          const jobId = `${{ env.JOB_ID }}`
+          let branch = "";
+          let failedSteps = "";
+          
+          if (event == "pull_request") {
+            branch = headRef
+          } else {
+            branch = ref.split("/")[2]
+          };
+          
+          commitString = `${{ github.event.head_commit.message }}`.split("/")[0];
+          
+          Object.keys(steps).forEach(function(key) {
+            if (steps[key]["conclusion"] != "success") {
+              failedSteps += String(key) + ": " + JSON.stringify(steps[key])
+            } else {
+            console.log(key)
+          }
+          });
+          
+          let message = `🔴 Workflow testing failed:
```

Comment:
You should decide, it is about `workflow` or `job`. Somewhere you write `workflow`, somewhere you write `job`. I suggest using 'job' everywhere.

Workflow testing failed -> Job failed

Reply:
Done